### PR TITLE
Index api consistency

### PR DESCRIPTION
**Pull request by SpacemanPaul**
### Reason for this pull request

1. API inconsistencies were identified during work on #1226.
2. redis-server was left out of Docker build in #1233, bypassing celery tests.

### Proposed changes

1. Regularise API inconsistencies
2. Add redis-server to docker image to allow celery tests to run

Note: The celery integration depends on ancient and security-compromised version of celery.  The general consensus seems to be that it is useless and nobody uses it, but I need to consult more widely before removing it all together.

### CODE REVIEW

File: datacube/index/postgres/_datasets.py [hunk 1]
```diff
@@ -116,9 +116,11 @@ def get_derived(self, id_):
         """
         Get all derived datasets
 
-        :param UUID id_: dataset id
+        :param Union[str,UUID] id_: dataset id
         :rtype: list[Dataset]
         """
+        if not isinstance(id_, UUID):
+            id_ = UUID(id_)
```
Comment (jeremyh) [inline datacube/index/postgres/_datasets.py:123]: Was this failing? I'm pretty sure the query works with both types. The other nearby methods like `has()` don't seem to convert their input UUID.

Test of the old code on both types:
```
In [3]: len(dc.index.datasets.get_derived("b069121e-8d32-48ed-b5eb-6335d2884bc9"))
Out[3]: 11

In [4]: from uuid import UUID

In [5]: len(dc.index.datasets.get_derived(UUID("b069121e-8d32-48ed-b5eb-6335d2884bc9")))
Out[5]: 11

In [6]: 
```
(... unfortunately it doesn't look like we have tests on those methods in the actual repo, at least not directly. But there are some integration tests in wider repos.)

Comment (SpacemanPaul) [reply datacube/index/postgres/_datasets.py:123]: Not failing per se, but the new code will catch the case where a string is passed in that is not a valid UUID.

**Review by jeremyh — APPROVED** [review]

File: datacube/index/postgres/_datasets.py (hunk 1 above)
Comment (jeremyh) [reply datacube/index/postgres/_datasets.py:123]: Ah, good point!

**Review by omad — COMMENTED** [review]
LGTM

**Pull request merged**